src/rest_vol.c

```diff
@@ -687,6 +727,7 @@ H5_rest_set_connection_information(void)
          * URL, username and password key-value pairs.
          */
         while (fgets(file_line, sizeof(file_line), config_file) != NULL) {
+            if ((file_line[0] == '#') || !strlen(file_line)) continue;
```

Comment:
This line should probably be moved below the variable declarations and then have have the variables get initialized after this line. Otherwise, compiling in ISO C mode will complain about variable declarations after a statement.

Reply:
Why not move them all the way "up" at the beginning of the `else` block? I wanted to avoid unnecessary parsing of lines with no interesting content.

Reply:
Sure, I understand that it's to avoid parsing any lines that are just comments. I'm just saying it should go after key and val are declared or C89 mode for C compilers will complain. Basically, just change this to:

const char *key;
const char *val;

if ((file_line[0] == '#') || !strlen(file_line)) continue;

key = strtok(file_line, " =\n");
val = strtok(NULL, " =\n");

That will satisfy ISO C compiling mode and also avoid tokenizing the line until after it's been checked for being a comment line. The only gotcha may be that some compilers might complain about not initializing a const variable and assigning to it later. You may need to remove the const in this case if so.

Reply:
Done.